File: src/java/Utils/OutputFileWritter.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package Utils;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author XPS 1645
 */
public class OutputFileWritter extends FileManager
{
    BufferedWriter myFileWritter;

    @Override
    public void OpenFile(String fileName)
    {
        FileWriter fstream = null;
        try {
            fstream = new FileWriter(fileName);
            myFileWritter = new BufferedWriter(fstream);
        } catch (IOException ex) {
            Logger.getLogger(OutputFileWritter.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    @Override
    public void CloseFile() throws IOException
    {
        myFileWritter.close();
    }

    public void WriteToFile(String currOutput) throws IOException
    {
        myFileWritter.write(currOutput);
        myFileWritter.newLine();
    }
}

File: src/c#/Utils/OutputFileWritter.cs
/*
 * created by: b farooq, poly montreal
 * on: 22 october, 2013
 * last edited by: b farooq, poly montreal
 * on: 22 october, 2013
 * summary: 
 * comments:
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace PopulationSynthesis.Utils
{
    class OutputFileWritter : FileManager
    {
        TextWriter myFileWritter;
        public override void OpenFile(string fileName)
        {
            myFileWritter = new StreamWriter(fileName);
        }
        public override void CloseFile()
        {
            myFileWritter.Close();
        }
        public void WriteToFile(string currOutput)
        {
            myFileWritter.WriteLine(currOutput);
        }
    }
}
